### PR TITLE
DOC: add roadmap

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #845

Copies the roadmap from the wiki page with some format tweaks for rst.

Updates conf.py settings

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

Inspection of the docs after compile.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
I've copied the header levels over, but they may need to be adjusted

File: docs/roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+`Issues <https://github.com/pysat/pysat/issues>`_ and
+`Projects <https://github.com/pysat/pysat/projects>`_.
+
+An item being on the roadmap does not mean that it will necessarily happen.
+During the implementation or testing periods we may discover issues that limit
+the feature.
+
+Generality
+----------
+pysat's data support is currently focused on space-science data sets, however,
```
Comment (aburrell) [inline docs/roadmap.rst:22]:
```suggestion
pysat's data support is currently focused on space-science data sets. However,
```

File: docs/roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+pysat's feature set works well on other types of data as well. Where appropriate,
```
Comment (aburrell) [inline docs/roadmap.rst:23]:
```suggestion
pysat's feature set also works well on other types of data. Where appropriate,
```

File: docs/roadmap.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+space-science specific features will be generalized for a wider audience.
+
+Data Support
+~~~~~~~~~~~~
+The Instrument class currently supports both pandas and xarray formats, covering
```
Comment (aburrell) [inline docs/roadmap.rst:28]:
```suggestion
The :py:class:`Instrument` class currently supports both :py:class:`pandas.DataFrame` and :py:class:`xarray.Dataset` formats, covering
```

File: docs/roadmap.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+one and multi-dimensional data that can be loaded into memory. Even larger data
```
Comment (aburrell) [inline docs/roadmap.rst:29]:
```suggestion
1D time-series and multi-dimensional data that can be loaded into memory. Even larger data
```

File: docs/roadmap.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+the additional complexity when working with multiple sources this may not always
+be possible. Long term, we intend on providing functionality that can merge a
+Constellation into a 'live' Instrument object for greatest compatibility.
+
+Metapackage
+~~~~~~~~~~~
+The minimal barriers to entry in open source software allows for a large
+variety of packages, each with its own approach to a problem. A disadvantage
+of the setup is the variety of packages may not have been developed with
+interoperability in mind, presenting challenges when attempting to combine
```
Comment (aburrell) [inline docs/roadmap.rst:52]:
```suggestion
of the setup is the variety of packages have been developed without
interoperability in mind, presenting challenges when attempting to combine
```

File: docs/roadmap.rst [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+these disparate packages towards a common goal. pysat's versatility when
+coupling to data sources may be used to connect these isolated packages
+together using pysat an intermediary. Once a package is connected to pysat
+then that functionality becomes available to all packages that incorporate
+pysat as a source. The value and functionality of this large scale pysat
+metapackage increases exponentially with every new connection.
+
+File Support
+~~~~~~~~~~~~
+pysat currently supports tracking both data and metadata, as well as the ability
```
Comment (aburrell) [inline docs/roadmap.rst:62]:
```suggestion
:py:mod:`pysat` currently supports tracking both data and metadata, as well as the ability
```

Here and elsewhere

File: docs/roadmap.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+to create netCDF4 files, and is capable of maintaining compliance with NASA's
+Space Physics Data Facility (SPDF/ISTP) formatting requirements for NASA
```
Comment (aburrell) [inline docs/roadmap.rst:64]: Recommend making this text a link to the formatting requirements.

**Review by aburrell — COMMENTED** [review]
Some grammar and formatting suggestions.

File: docs/roadmap.rst [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+satellite missions. Support for creating different types of files, as well as
```
Comment (jklenzing) [inline docs/roadmap.rst:65]:
```suggestion
`Space Physics Data Facility (SPDF) <https://spdf.gsfc.nasa.gov/sp_use_of_cdf.html>`_
formatting requirements for NASA
satellite missions. Support for creating different types of files, as well as
```

**Review by aburrell — COMMENTED** [review]
Need to check line length and replace 'pysat' with ':py:mod:`pysat`' everywhere in the roadmap.rst file.

File: docs/roadmap.rst [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+tasks associated with this roadmap may be found within the posted
+`Issues <https://github.com/pysat/pysat/issues>`_ and
+`Projects <https://github.com/pysat/pysat/projects>`_.
+
+An item being on the roadmap does not necessarily mean that it will happen.
+During the implementation or testing periods we may discover issues that limit
+the feature.
+
+Generality
+----------
```
Comment (aburrell) [inline docs/roadmap.rst:22]:
```suggestion
```

Recommend removing this header.  The word is not used correctly (from what I could see looking up the definition) and the text goes well under the top level header.

File: docs/roadmap.rst [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+Data support with :py:mod:`pysat` is currently focused on space-science data
+sets. However, the features within the module also work well on other types of
+data. Where appropriate, space-science specific features will be generalized for
+a wider audience.
+
+Data Support
+~~~~~~~~~~~~
```
Comment (aburrell) [inline docs/roadmap.rst:29]:
```suggestion
-------------------
```

And below replace all `~` with `-`

**Review by aburrell — COMMENTED** [review]
Checked the links and formatting.  The headers were the only thing not working as they should be, so this will fix them.

File: docs/roadmap.rst [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+During the implementation or testing periods we may discover issues that limit
+the feature.
+
+Data support with :py:mod:`pysat` is currently focused on space-science data
+sets. However, the features within the module also work well on other types of
+data. Where appropriate, space-science specific features will be generalized for
+a wider audience.
+
+Data Support
+-------------------
```
Comment (jklenzing) [inline docs/roadmap.rst:27]:
```suggestion
Data Support
------------
```

File: docs/roadmap.rst [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+1D time-series and multi-dimensional data that can be loaded into memory. Even
+larger data sets would require that :py:mod:`pysat` integrate a data format such
+as Dask. To cover the needs of any potential user, an ideal solution would be
+for :py:mod:`pysat` to implement a clear public mechanism for users to add their
+own data formats. Commonalities observed after integrating :py:mod:`dask`,
+:py:mod:`pandas`, and :py:mod:`xarray` should provide a viable path forward for
+this generalization.
+
+Multiple Data Sources
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (aburrell) [inline docs/roadmap.rst:39]: All of these headers need to be "-"

File: docs/roadmap.rst [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+enhanced to support a broader array of research areas.
+
+Data Iteration
+~~~~~~~~~~~~~~
+:py:mod:`pysat` currently features orbit iteration, a feature that transparently
+provides complete orbits (across day/file breaks) calculated in real time. A
+variety of orbit types are supported, each of which maps to a method looking for
+a particular signal in the data to trigger upon. However, the current variety of
+orbit types is insufficient to address community needs. The underlying class is
+capable of iterating over a wider variety conditions though this type of
```
Comment (rstoneback) [inline docs/roadmap.rst:81]:
```suggestion
capable of iterating over a wider variety of conditions though this type of
```

**Comment by jklenzing** [comment]
@aburrell, @rstoneback, any chance this pull could be merged into the RC?

**Review by rstoneback — APPROVED** [review]

**Comment by rstoneback** [comment]
> @aburrell, @rstoneback, any chance this pull could be merged into the RC?

Sounds good to me.

**Pull request merged**